net/libfab/libfab.c

```diff
@@ -0,0 +1,434 @@
+/* -*- C -*- */
+/*
+ * Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For any questions about this software or licensing,
+ * please email opensource@seagate.com or cortx-questions@seagate.com.
+ *
+ */
+
+
+/**
+ * Mapping for m0net-->libfabric API
+ * xo_dom_init                    =  fi_getinfo, fi_fabric, fi_domain
+ * xo_dom_fini                    =  fi_close, // free resources
+ * xo_tm_init                     =  // Empty function ? fi_send
+ * xo_tm_confine                  =
+ * xo_tm_start                    =  // fi_send
+ * xo_tm_stop                     =  fi_cancel
+ * xo_tm_fini                     =  // Empty function
+ * xo_end_point_create            =  fi_endpoint, fi_pep, fi_av, fi_cq, fi_cntr, fi_eq, fi_bind(av/cq/cntr/eq), fi_pep_bind
+ * xo_buf_register                =  fi_mr_reg, fi_mr_desc, fi_mr_key, fi_mr_bind, fi_mr_enable
+ * xo_buf_deregister              =  fi_close
+ * xo_buf_add                     =  fi_send/fi_recv
+ * xo_buf_del                     =  fi_cancel 
+ * xo_bev_deliver_sync            =  Is it needed?
+ * xo_bev_deliver_all             =  Is it needed?
+ * xo_bev_pending                 =  Is it needed?
+ * xo_bev_notify                  =  Is it needed?
+ * xo_get_max_buffer_size         =  // need to define new API
+ * xo_get_max_buffer_segment_size =  // need to define new functions
+ * xo_get_max_buffer_segments     =  // need to define new functions
+ * xo_get_max_buffer_desc_size    =  
+ * 
+ */
+
+/**
+ * @addtogroup netlibfab
+ *
+ * @{
+ */
+
+#define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_NET
+#include "lib/trace.h"
+
+#include "net/net.h"
+#include "lib/bitmap.h"
+#include "lib/chan.h"
+#include "lib/memory.h"
+#include "libfab_internal.h"
+#include <errno.h>
+
+
+/** Network transfer machine */
+struct transfer_ma {
```

Comment:
rename 
`transfer_ma `
to
`libfab_transfer_ma`
?

Reply:
we will update this PR#244

Reply:
transfer_ma is fine, this is a local structure. In fact, net/sock/sock.c call this simply "struct ma".